core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -1212,7 +1246,7 @@ public TableMetadata build() {
           ImmutableList.copyOf(newSnapshotLog),
           ImmutableList.copyOf(metadataHistory),
           ImmutableMap.copyOf(refs),
-          discardChanges ? ImmutableList.of() : ImmutableList.copyOf(changes)
+          discardChanges || base == null ? ImmutableList.of() : ImmutableList.copyOf(changes)
```

Comment:
I'm not sure about this. If base is `null`, aren't all of the changes still valid?

Tests pass without this change, so I would probably not include it. What was the reason for adding this?

Reply:
`TestLoadTableResponse.testHasOnlyKnownFields()` was failing, so my intent was to make it behave like `newTableMetadata` which was setting changes to empty, though probably my misunderstanding. I'll remove this and fix what's going on with that test.

Reply:
I made this change